### PR TITLE
internationalized models validation error messages

### PR DESCRIPTION
**Pull request by DreamVenerable**
# 🔗 Issue
#153 

# ✍️ Description

Added internationalization to model validation error messages

# 📷 Screenshots/Demos

### CODE REVIEW

**Comment by DreamVenerable** [comment]
Can someone review this and see if I need to change anything?

**Comment by DreamVenerable** [comment]
test is failing for main, do I need to fix this to proceed?

**Comment by kasugaijin** [comment]
@DreamVenerable I think it's a flaky test. Not an issue with this PR, as it sometimes fail on other branches, too. 
Did you manually check these errors being rendered for each model?

**Comment by DreamVenerable** [comment]
I checked these errors on the console, but not on render. I'll go check them right now

**Comment by DreamVenerable** [comment]
@kasugaijin I can't run latest migration, which is stopping me from opening the web app on localhost 


Screenshot 2024-01-08 at 12 00 57 AM

**Comment by kasugaijin** [comment]
Try merging main branch into yours then see if that helps.

**Comment by kasugaijin** [comment]
> I checked these errors on the console, but not on render. I'll go check them right now

Ok great. No need to check every error but just check that each mod it getting it's errors from the yml file. Change the yml text temporarily to make sure if you want an easy way to check.

**Comment by DreamVenerable** [comment]
> Try merging main branch into yours then see if that helps.

It didn't help, I even tried on main and I also can't run migration there

**Comment by kasugaijin** [comment]
OK I think I know why. This merge into main https://github.com/rubyforgood/pet-rescue/commit/441a2be50c649ca48accd207afd478ed9e59abd3#diff-9e93b7ccff26ecc886221d13724d22344e7bcf2c14bccdf8ef7c6fa9fb659543 removed the strong_migrations gem which is what provides the `safety_assured` method. Can you please re-write that failing migration so it is not using `safety_assured`? That should then get it to migrate. You can also keep and commit that change so we reflect it in main branch.

**Comment by DreamVenerable** [comment]
Alright, I'll do that, thanks for the help!

**Comment by DreamVenerable** [comment]
@kasugaijin I checked and indeed there were some errors that got their start cut off or didn't reflect changes in the yml. 

Now it should be all good I hope

File: config/locales/en.yml [hunk 1]
```diff
@@ -52,6 +52,187 @@ en:
...
+            sex:  
+              blank: "Sex can't be blank"
+            species:  
+              blank: "Species can't be blank"
+            placement_type: 
+              blank: "Placement type can't be blank"
+            weight_from: 
+              blank: "Weight From can't be blank"
+              not_a_number: "Weight From is not a number"
+            weight_to:
```
Comment (kasugaijin) [inline config/locales/en.yml:202]: There should be another message here related to the error where `weight_to` is less than `weight_from`. You can test in the UI by adding a lower number in the `weight_to` field compared to `weight_from` field.

**Comment by kasugaijin** [comment]
Looking good so far. In case you have not, I'd recommend checking all models and their validations to make sure you have captured the potential errors :) Case in point there is a validation on Pet to make sure `weight_to` is  greater number than `weight_from` that was missing in the .yml

**Comment by kasugaijin** [comment]
Hey @DreamVenerable are you happy with this PR now? No rush!

**Comment by DreamVenerable** [comment]
@kasugaijin First time using the review feature yesterday, but it seems like you don't see it cuz its pending?

Screenshot 2024-01-10 at 12 20 16 AM

Anyways, yes for the most part this PR looks good enough

**Comment by kasugaijin** [comment]
@DreamVenerable Oh sorry I did not see that or get notified for it. I think we can keep the validation. There's no guarantee the JS validation will work/remain in use in the future, so a back end validation is a good thing.

**Comment by DreamVenerable** [comment]
Alright then!

Is this PR ready to be merged?

**Review by kasugaijin — APPROVED** [review]
lgtm

**Pull request merged**